packages/clay-form/src/SelectWithOption.tsx

```diff
@@ -12,17 +12,35 @@ interface IProps extends React.ComponentProps<typeof Select> {
 	/**
 	 * Options of the select.
 	 */
-	options: Array<React.ComponentProps<typeof Select.Option>>;
+	options: Array<
+		(
+			| React.ComponentProps<typeof Select.Option>
+			| React.ComponentProps<typeof Select.OptGroup>) & {
+			options?: Array<React.ComponentProps<typeof Select.Option>>;
+			type?: 'group';
```

Comment:
@bryceosterhaus just a curiosity can you tell if it is possible to create nested groups?

Reply:
@matuzalemsteles its not possible with our implementation and it also doesnt seem to be in the HTML spec for `<optgroup>`

Reply:
Sounds good, np.